### PR TITLE
Sample snippet for Google Cloud Storage including a test fixture

### PR DESCRIPTION
**Pull request by jskeet**
This is essentially a "getting started with snippets" change to unblock
further snippet development. Some notes have been included in CONTRIBUTING.md
but they are likely to require more work later.

### CODE REVIEW

**Comment by jskeet** [comment]
(I'm going to start adding Datastore snippets in a similar way, to take the place of the demo code that's now been deleted.)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -23,6 +23,102 @@ we'll be able to accept your pull requests.
...
+  the [API Manager](https://console.developers.google.com/apis/library)
+- Download and install the [https://cloud.google.com/sdk/](Cloud SDK)
+- Run `gcloud auth login` to set up default application credentials
+- Set the `TEST_PROJECT` environment variable to the name of your project.
+  To run the tests from a command line, you can just set them directly
+  and run the tests. To run the tests from Visual Studio, you should set
+  the environment variable permanently, and restart Visual Studio if
+  it's already running.
+- Build all the code, either from Visual Studio, or from the command line
+  with `dnu`/`dnx`.
```
Comment (csells) [inline CONTRIBUTING.md]: would it be possible to see the specific sequence of commands here that build and test the unit tests and that build and test the snippet tests?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -23,6 +23,102 @@ we'll be able to accept your pull requests.
...
+---
+
+Code in the `snippets` directory can be marked up to be used in the
+API reference documentation. The
+`Google.GCloud.Tools.GenerateSnippetMarkdown` tool (in the `tools`
+directory) extract snippets from projects underneath `snippets`. To
+add a snippet, simply include a comment on a line on its own at the
+start of the snippet, of the form:
+
+    // <method-name>
```
Comment (csells) [inline CONTRIBUTING.md]: example?

Comment (jskeet) [inline CONTRIBUTING.md]: Added a full example below.

**Comment by jskeet** [comment]
I'll have to think about how to distinguish between overloads for our REST API wrappers, e.g. `DownloadObject(string, string, Stream, DownloadObjectOptions, IProgress<IDownloadProgress>` and `DownloadObject(Object, Stream, DownloadObjectOptions, IProgress<IDownloadProgress>`.

(The snippet extractor needs to be reworked a bit to handle these snippets too.) For the moment, we can all write snippets assuming it'll just work, and I'll adjust them appropriately when we've addressed that.

**Pull request merged**